rpc-client-api/src/filter.rs

```diff
@@ -78,14 +72,6 @@ impl RpcFilterType {
             RpcFilterType::TokenAccountState => Ok(()),
         }
     }
-
-    pub fn allows(&self, account: &AccountSharedData) -> bool {
```

Comment:
Unfortunately, this is part of the public interface of a client crate, and while I don't know of a client use-case for it, we shouldn't remove it without first deprecating. Can you please restore and add a `#[deprecated]` tag?

In order to still remove the spl_token_2022 dependency now, we can import `valid_account_data()` from the `solana-inline-spl` crate, which is at least much smaller than `spl-token-2022`. Then we can remove that entirely in the future.